src/api/commands/listingitem/ListingItemPostCommand.ts

```diff
@@ -60,17 +59,14 @@ export class ListingItemPostCommand implements RpcCommandInterface<ListingItem>
                 marketId = defaultMarket.id;
             }
 
-            // convert listing Item template object to camelcaseObject
-            const newListingTemplate = camelcaseObject(itemTemplate);
-
-            const addItemMessage = await this.listingItemFactory.get({
-                hash: ObjectHash.getHash(newListingTemplate),
-                marketId: data.params[1] || marketId,
-                listingItemTemplateId: newListingTemplate.id,
-                information: newListingTemplate.itemInformation || {},
-                payment:  newListingTemplate.paymentInformation || {},
-                messaging: newListingTemplate.messagingInformation || {}
-            } as ListingItemMessage);
+            // get the ListingItemMessage from listing item template
+            const addItemMessage = await this.listingItemFactory.getMessage({
+                hash: ObjectHash.getHash(itemTemplate),
+                listingItemTemplateId: itemTemplate.id,
+                information: itemTemplate.itemInformation || {},
+                payment:  itemTemplate.paymentInformation || {},
+                messaging: itemTemplate.messagingInformation || {}
+            } as ListingItemMessage, marketId);
 
```

Comment:
your passing the requested type allready to the factory, so whats the point of the factory?